core/src/main/AndroidManifest.xml

```diff
@@ -11,7 +11,8 @@
 
   <!-- Device with versions >= Pie need this permission -->
   <uses-permission android:name="android.permission.FOREGROUND_SERVICE" />
-
+  <uses-permission android:name="android.permission.QUERY_ALL_PACKAGES"
```

Comment:
We don't need this permission.

Reply:
its required to overcome wifi issue on android 30, otherwise its running foreover

https://stackoverflow.com/questions/62345805/namenotfoundexception-when-calling-getpackageinfo-on-android-11

Reply:
We don't need to query or interact with all installed apps on a device. If we wanna query and interact with specific packages then we can do by this
````
<manifest package="com.example.game">
  <queries>
    <!-- Specific apps you interact with, eg: -->
    <package android:name="com.example.store" />
    <package android:name="com.example.service" />
````

Reply:
So, after all, we need this permission!

Reply:
Yes

Reply:
I just tried this on latest commit but it won't work now android 30 running forever we must revert to android:name="android.permission.QUERY_ALL_PACKAGES"